### PR TITLE
Improvements for uninstall

### PR DESCRIPTION
**Pull request by mbtools**
Uninstall now catches exceptions and tries up to 3 times to remove an object. Exceptions are only raised if after 3 tries some objects still remain. 

Tested with https://github.com/larshp/abapPGP

Closes https://github.com/abapGit/abapGit/issues/379 and https://github.com/abapGit/abapGit/issues/3949, and probably #3228

PS: In certain cases with cyclic dependencies it won't be possible to automatically delete.

### CODE REVIEW

**Comment by larshp** [comment]
@mbtools check conflicts

File: src/objects/zcl_abapgit_objects.clas.abap [hunk 1]
```diff
@@ -495,28 +496,43 @@ CLASS ZCL_ABAPGIT_OBJECTS IMPLEMENTATION.
...
-
       CATCH zcx_abapgit_exception INTO lx_error.
         zcl_abapgit_default_transport=>get_instance( )->reset( ).
         RAISE EXCEPTION lx_error.
     ENDTRY.
 
+    DO 3 TIMES.
+      LOOP AT lt_tadir ASSIGNING <ls_tadir>.
+        lv_count = lv_count + 1.
+        li_progress->show( iv_current = lv_count
```
Comment (larshp) [inline src/objects/zcl_abapgit_objects.clas.abap:507]: line 492: `li_progress = zcl_abapgit_progress=>get_instance( lines( lt_tadir ) ).`

perhaps move the progress bar initialization to after the DO and before LOOP

Comment (mbtools) [reply src/objects/zcl_abapgit_objects.clas.abap:507]: when an object is successfully deleted, the `lt_tadir` entry is removed. `lv_count` should therefore count the entries properly. The progress looked ok during tests.

Comment (larshp) [reply src/objects/zcl_abapgit_objects.clas.abap:507]: worst case scenario, everything fails during deletion, then lv_count would be 3x the size of `lines( lt_tadir )` ?

I think moving the progress indicator to line 520 will make it work

**Comment by larshp** [comment]
thanks 👍

**Pull request merged**